test/e2e/storage/testsuites/provisioning.go

```diff
@@ -288,6 +307,13 @@ func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, cla
 		By("checking the created volume is readable and retains data")
 		runInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, "grep 'hello world' /mnt/test/data", t.NodeSelector, t.ExpectUnschedulable)
 	}
+
+	if claim.Spec.DataSource != nil {
+		By("checking the created volume whether has the pre-populated data")
+		command := fmt.Sprintf("grep '%s' /mnt/test/initialData", claim.Namespace)
+		runInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, command, t.NodeSelector, t.ExpectUnschedulable)
+	}
+
```

Comment:
This check here should be in the "should provision storage with snapshot data source" test case, not here.